tensorflow/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -0,0 +1,4547 @@
+      ScopedActivationDescriptor& activation_descriptor) {
+    uint64 hashValue = tensorflow::Hash64("BatchNormActivationBackward");
+
+    hashValue = tensorflow::Hash64Combine(
+        hashValue, tensorflow::hash<miopenHandle_t>()(miopen_handle));
+
+    hashValue =
+        tensorflow::Hash64Combine(hashValue, GetHashValue(input_descriptor));
+
+    hashValue = tensorflow::Hash64Combine(
+        hashValue, GetHashValue(scale_offset_mean_variance_descriptor));
+
+    hashValue = tensorflow::Hash64Combine(hashValue,
+                                          activation_descriptor.GetHashValue());
+    return hashValue;
+  }
+
+ private:
+  const int k_batchnorm_op_idx = 0;
+  const int k_actv_op_idx = 1;
+
+  SE_DISALLOW_COPY_AND_ASSIGN(ScopedFusionPlanBatchNormActivationBackward);
+};
+
+namespace {
+miopenDataType_t ToMIOpenDataType(
+    dnn::DataType data_type,
+    dnn::DataLayout data_layout = dnn::DataLayout::kBatchDepthYX) {
+  switch (data_type) {
+    case dnn::DataType::kFloat:
+      return miopenFloat;
+    case dnn::DataType::kHalf:
+      return miopenHalf;
+    case dnn::DataType::kDouble:
+    default:
+      LOG(FATAL) << "Invalid DNN data type: " << static_cast<int>(data_type);
+  }
+}
+
+miopenDataType_t ToMIOpenDataType(dnn::DataType data_type,
+                                  dnn::FilterLayout filter_layout) {
+  return ToMIOpenDataType(data_type);
+}
+
+miopenRNNInputMode_t ToMIOpenRnnInputMode(dnn::RnnInputMode input_mode) {
+  switch (input_mode) {
+    case dnn::RnnInputMode::kRnnLinearSkip:
+      return miopenRNNlinear;
+    case dnn::RnnInputMode::kRnnSkipInput:
+      return miopenRNNskip;
+    default:
+      LOG(FATAL) << "Invalid RNN input mode: " << static_cast<int>(input_mode);
+  }
+}
+
+miopenRNNDirectionMode_t ToMIOpenRnnDirectionMode(
+    dnn::RnnDirectionMode direction_mode) {
+  switch (direction_mode) {
+    case dnn::RnnDirectionMode::kRnnUnidirectional:
+      return miopenRNNunidirection;
+    case dnn::RnnDirectionMode::kRnnBidirectional:
+      return miopenRNNbidirection;
+    default:
+      LOG(FATAL) << "Invalid RNN direction mode: "
+                 << static_cast<int>(direction_mode);
+  }
+}
+
+miopenRNNMode_t ToMIOpenRnnMode(dnn::RnnMode rnn_mode) {
+  switch (rnn_mode) {
+    case dnn::RnnMode::kRnnRelu:
+      return miopenRNNRELU;
+    case dnn::RnnMode::kRnnTanh:
+      return miopenRNNTANH;
+    case dnn::RnnMode::kRnnLstm:
+      return miopenLSTM;
+    case dnn::RnnMode::kRnnGru:
+      return miopenGRU;
+    default:
+      LOG(FATAL) << "Invalid RNN Mode: " << static_cast<int>(rnn_mode);
+  }
+}
+
+int MIOpenDataTypeToByteSize(miopenDataType_t data_type) {
+  switch (data_type) {
+    case miopenFloat:
+      return sizeof(float);
+    case miopenHalf:
+      return sizeof(Eigen::half);
+    default:
+      LOG(FATAL) << "Invalid DNN data type: " << static_cast<int>(data_type);
+  }
+}
+
+template <typename Base>
+class MixinBase : public Base
```

Comment:
updated the name of the macro to RETURN_IF_MIOPEN_ERROR, but I cannot have it return a Status like its CUDA equivalent. This is because, all the invocations of this macro are within constructors/destructors. So would like to leave the return statements as is.

Reply:
Would it be possible to use cuda_dnn's approach of factory methods returning a StatusOr?

Reply:
I would like to leave this as is, because as indicated earlier those macros are invoked within constructors/destructors on the ROCm side.